### PR TITLE
rpc: add bestblock grpc.

### PR DESCRIPTION
**Pull request by dnldd**
requested by #1085

### CODE REVIEW

File: rpc/api.proto [hunk 1]
```diff
@@ -57,6 +57,7 @@ service WalletService {
 	rpc SignMessages (SignMessagesRequest) returns (SignMessagesResponse);
 	rpc ValidateAddress (ValidateAddressRequest) returns (ValidateAddressResponse);
 	rpc CommittedTickets (CommittedTicketsRequest) returns (CommittedTicketsResponse);
+	rpc BestBlock (BestBlockRequest) returns (BestBlockResponse);
```
Comment (jrick) [inline rpc/api.proto]: Move this under the queries section

File: rpc/api.proto [hunk 2]
```diff
@@ -901,3 +902,11 @@ message CommittedTicketsResponse {
 	}
 	repeated TicketAddress ticketAddresses = 1;
 }
+
+message BestBlockRequest {
+}
+
+message BestBlockResponse {
+	int64 height = 1;
```
Comment (jrick) [inline rpc/api.proto]: uint32 (normally i would ask for int32 in Go code, but unsigned is preferred for performance reasons in protobuf).

File: rpc/documentation/api.md [hunk 3]
```diff
@@ -1777,6 +1778,19 @@ addresses for which the largest commitment is controlled by this wallet.
...
+
+The `BestBlock` method returns the block height and height of the best block on
+the main chain.
+
+**Request:** `BestBlockRequest`
+
+**Response:** `BestBlockResponse`
+
+- `bytes hash`: The hash of the best block.
+- `int64 height`: The height of the best block.
```
Comment (jrick) [inline rpc/documentation/api.md]: add a newline between these for consistency

File: rpc/rpcserver/server.go [hunk 4]
```diff
@@ -2750,3 +2750,27 @@ func (s *decodeMessageServer) DecodeRawTransaction(ctx context.Context, req *pb.
 
 	return resp, nil
 }
+
+func (s *walletServer) BestBlock(ctx context.Context, req *pb.BestBlockRequest) (
+	*pb.BestBlockResponse, error) {
```
Comment (jrick) [inline rpc/rpcserver/server.go]: keep on same line

File: rpc/rpcserver/server.go [hunk 5]
```diff
@@ -2750,3 +2750,27 @@ func (s *decodeMessageServer) DecodeRawTransaction(ctx context.Context, req *pb.
...
+	n, err := s.requireNetworkBackend()
```
Comment (jrick) [inline rpc/rpcserver/server.go]: We should be querying the wallet's best block, not the network's.  This does not require access to the network backend.  Use https://godoc.org/github.com/decred/dcrwallet/wallet#Wallet.MainChainTip for this

File: rpc/rpcserver/server.go [hunk 6]
```diff
@@ -2750,3 +2750,27 @@ func (s *decodeMessageServer) DecodeRawTransaction(ctx context.Context, req *pb.
...
+		return nil, translateError(err)
+	}
+
+	hash, height, err := chainClient.GetBestBlock()
+	if err != nil {
+		return nil, translateError(err)
+	}
+
+	resp := &pb.BestBlockResponse{
+		Hash:   hash.CloneBytes(),
```
Comment (jrick) [inline rpc/rpcserver/server.go]: a copy/clone is unnecessary

Comment (dnldd) [reply rpc/rpcserver/server.go]: Elaborate, `hashCloneBytes()` is equivalent to `[]bytes(hash.String())`, or should I be returning the hash string instead?

Comment (jrick) [reply rpc/rpcserver/server.go]: `hash[:]`

File: rpc/documentation/api.md [hunk 7]
```diff
@@ -1777,6 +1778,20 @@ addresses for which the largest commitment is controlled by this wallet.
...
+The `BestBlock` method returns the block height and height of the best block on
+the main chain.
+
+**Request:** `BestBlockRequest`
+
+**Response:** `BestBlockResponse`
+
+- `bytes hash`: The hash of the best block.
+
+- `int64 height`: The height of the best block.
```
Comment (jrick) [inline rpc/documentation/api.md]: uint32

**Pull request merged**